pkg/retry/retry_test.go

```diff
@@ -335,3 +335,100 @@ func TestOnError_RetryCountResetTimeout(t *testing.T) {
 		require.Contains(t, line, expectedPrefix)
 	}
 }
+
+// assert that a negative retry limit continually calls workFn
+func TestOnError_NegativeRetryLimit(t *testing.T) {
+	t.Skip("TODO_TECHDEBT: this test should pass but contains a race condition around the logOutput buffer")
```

Comment:
@bryanchriswhite @ezeike So is the decision to fix this or merge it in as is?

Reply:
Merge in as is. Tests confirmed passing locally without the -race flag.

Reply:
Let's skip any tests which will fail with the `-race` flag present as we should be using it in CI and in general to mitigate adding race conditions.

Reply:
@bryanchriswhite I know you participated in contributing this PR, but could you also review and approve it assuming it looked good to you?

@ezeike Could you please review [1] to run E2E tests.

[1] https://github.com/pokt-network/poktroll/blob/main/.github/pull_request_template.md